### PR TITLE
Rust 2018 transition

### PR DESCRIPTION
**Pull request by XVilka**
Updating to the [Rust 2018 edition](https://doc.rust-lang.org/edition-guide/rust-2018/index.html).

### CODE REVIEW

**Comment by XVilka** [comment]
Ping?

**Comment by waywardmonkeys** [comment]
The minimum Rust version has now been bumped to 1.37 for the 0.5 release. I'd like to visit things related to Rust 2018 in a bit-by-bit fashion, with separate commits for each type of change.

**Comment by XVilka** [comment]
Will split into separate changes once PR is green.

**Comment by XVilka** [comment]
@waywardmonkeys ok, it is done - green, split and ready for review!

**Comment by jmcomets** [comment]
Is there a way we could add lints for these? I'm specifically thinking of things like the `use crate::` syntax that tend to creep back.

**Comment by XVilka** [comment]
@jmcomets there are two things that can be done - [setting up `cargo fmt`](https://github.com/bluss/petgraph/issues/269) and running `cargo clippy` (there are many warnings btw).

File: Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ debug = true
 
 [dependencies]
 fixedbitset = { version = "0.1.4" }
-quickcheck = { optional = true, version = "0.8", default-features = false }
+quickcheck = { optional = true, version = "0.8.5", default-features = false }
```
Comment (waywardmonkeys) [inline Cargo.toml]: Does this need to be 0.8.5? or will any 0.8 work?

Comment (XVilka) [reply Cargo.toml]: Yes, 0.8 will work too. I just saw that there was a quickcheck version bump, but not to the latest one. I don't see why not to bump to the latest quickcheck for the new release though. You can drop this commit if you don't want to.

Comment (waywardmonkeys) [reply Cargo.toml]: Okay. If you can drop that commit from this PR, we can get this landed.

**Comment by waywardmonkeys** [comment]
Thanks!

**Pull request merged**